src/Processors/Transforms/Streaming/AggregatingHelper.cpp

```diff
@@ -13,7 +13,18 @@ namespace
 {
 Chunk mergeBlocksToChunk(BlocksList && blocks)
 {
-    Chunk merged_chunk;
+    size_t total_rows = 0;
+    for (auto & block : blocks)
+        total_rows += block.rows();
+
+    if (total_rows == 0)
+        return {};
```

Comment:
We shall always return chunk with columns ? empty chunk basically means tear down the whole pipeline

Reply:
When return {}, means no aggregated result, we don't push it into output